src/plugin/places/placeshidecmd.js

```diff
@@ -0,0 +1,129 @@
+goog.provide('plugin.places.PlacesHide');
+
+goog.require('os.command.ICommand');
+goog.require('os.structs.TriState');
+goog.require('plugin.places');
+
+
+
+/**
+ * Command for hiding all places
+ * @implements {os.command.ICommand}
+ * @constructor
+ */
+plugin.places.PlacesHide = function() {
+  /**
+   * Features that visible at time of execution
+   * @type {Array}
+   * @private
+   */
+  this.visibleNodes_ = [];
+};
+
+
+/**
+ * @inheritDoc
+ */
+plugin.places.PlacesHide.prototype.isAsync = false;
+
+
+/**
+ * @inheritDoc
+ */
+plugin.places.PlacesHide.prototype.title = 'Hide Places';
+
+
+/**
+ * @inheritDoc
+ */
+plugin.places.PlacesHide.prototype.details = null;
+
+
+/**
+ * @inheritDoc
+ */
+plugin.places.PlacesHide.prototype.state = os.command.State.READY;
+
+
+/**
+ * @inheritDoc
+ */
+plugin.places.PlacesHide.prototype.execute = function() {
+  this.state = os.command.State.EXECUTING;
+  var rootNode = plugin.places.PlacesManager.getInstance().getPlacesRoot();
```

Comment:
This should clear `this.visibleNodes_` first, or it may result in expected behavior with undo/redo. It will also infinitely grow the list on each `execute` call.